lib/_http_incoming.js

```diff
@@ -1,7 +1,13 @@
 'use strict';
 
+const binding = process.binding('http_parser');
+const knownHttpHeaders = binding.knownHttpHeaders;
+const lowerCaseHttpHeaderLookup = Object.create(null);
 const util = require('util');
 const Stream = require('stream');
+for (var i = 0; i < knownHttpHeaders.length; i++) {
+  lowerCaseHttpHeaderLookup[knownHttpHeaders[i][1]] = knownHttpHeaders[i][2];
```

Comment:
I think it would be better to create an object literal via dynamic code generation to avoid the hidden classes created by the addition of these properties.

For example:

```js
const lowerCaseHttpHeaderLookup = (function() {
  var code = 'return Object.create(null, {';
  for (var i = 0; i < knownHttpHeaders.length; ++i) {
    const origName = knownHttpHeaders[i][1];
    const lowName = knownHttpHeaders[i][2];
    code += `${JSON.stringify(origName)}: {`;
    code += `value: ${JSON.stringify(lowName)}, enumerable: true`;
    code += `},`;
  }
  code += '})';
  return new Function(code)();
})();
```